mlir/lib/Dialect/Rock/Transforms/AttentionToGridwise.cpp

```diff
@@ -0,0 +1,179 @@
+//===- GemmToGridwise.cpp - Rock GEMM implementation ------------===//
+//
+// Copyright 2022 Advanced Micro Devices.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ============================================================
+//
+// This pass converts rock.attention into the appropriate
+// rock.gridwise_attention adding padding and group dimensions if needed.
+//
+//===-----------------------------------------------------===//
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Rock/IR/GemmSize.h"
+#include "mlir/Dialect/Rock/IR/Rock.h"
+#include "mlir/Dialect/Rock/IR/TransformMapBuilder.h"
+#include "mlir/Dialect/Rock/Passes.h"
+#include "mlir/Dialect/Rock/Tuning/GridwiseGemmParams.h"
+#include "mlir/Dialect/Rock/utility/AmdArchDb.h"
+#include "mlir/Dialect/Rock/utility/loweringUtils.h"
+
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/Support/LogicalResult.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+#include "llvm/Support/Debug.h"
+#include <memory>
+
+namespace mlir {
+namespace rock {
+#define GEN_PASS_DEF_ROCKATTENTIONTOGRIDWISEPASS
+#include "mlir/Dialect/Rock/Passes.h.inc"
+} // namespace rock
+} // namespace mlir
+
+#define DEBUG_TYPE "rock-attention-to-gridwise"
+
+using namespace mlir;
+using namespace mlir::rock;
+
+namespace {
+class RockAttentionToGridwisePass
+    : public rock::impl::RockAttentionToGridwisePassBase<
+          RockAttentionToGridwisePass> {
+  void runOnOperation() override;
+};
+
+struct AttentionRewritePattern : public OpConversionPattern<AttentionOp> {
+  using OpConversionPattern<AttentionOp>::OpConversionPattern;
+  LogicalResult matchAndRewrite(AttentionOp op, AttentionOpAdaptor adaptor,
+                                ConversionPatternRewriter &b) const override;
+};
+} // end namespace
+
+static Attribute getTuningParams(ConversionPatternRewriter &rw,
```

Comment:
This goes in `affix-tuning-params`, I claim

Reply:
yea -- as much as I dont like that pass , you are right in terms of consistency -- hence I ll do the movement.

Something for the future,
It;d probably better to ops 'own' the tuning params rather than a pass doing it for them. IIRC, you had some thoughts about removing that pass altogether.. anyway I ll park this conversation for later. (now is not the time)

Reply:
Yeah, I'm with you on having ops own the tuning parameters and having a `addHeuristicTuningParams(op)` and `computeDerivedTuningParemeters(op)` function ... probably with making the op require the tuning parameter attributes.

But that's a future thing.